src/Compatibility/Core/src/Windows/CellControl.cs

```diff
@@ -362,15 +355,8 @@ void SetCell(object newContext)
 						sameTemplate = oldTemplate == template;
 					}
 				}
-				// 🚀 if there is no datatemplateselector, we now verify if the old cell
-				// was a groupheader and whether the new one is as well.
-				// Again, this is only to verify we can reuse this cell
-				else if (wasGroupHeader.HasValue)
-				{
-					sameTemplate = wasGroupHeader == isGroupHeader;
-				}
 
-				// reuse cell
+				// Reuse cell
```

Comment:
Fix ListView grouping not working on Windows
https://github.com/xamarin/Xamarin.Forms/pull/14653